### PR TITLE
karmada operator: add karmada operator chart

### PR DESCRIPTION
**Pull request by calvin0327**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
We need multiple way to install karmada operator, like helm chart:

here is test result:

```shell
helm install karmada-operator -n karmada-system  --create-namespace  ./charts/karmada-operator --debug

NAME: karmada-operator
LAST DEPLOYED: Thu May 11 19:23:56 2023
NAMESPACE: karmada-system
STATUS: deployed
REVISION: 1
TEST SUITE: None
USER-SUPPLIED VALUES:
{}

COMPUTED VALUES:
common:
  exampleValue: common-chart
  global:
    imagePullSecrets: []
    imageRegistry: ""
global:
  imagePullSecrets: []
  imageRegistry: ""
operator:
  affinity: {}
  image:
    pullPolicy: IfNotPresent
    pullSecrets: []
    registry: docker.io
    repository: karmada/karmada-operator
    tag: latest
  labels: {}
  nodeSelector: {}
  podAnnotations: {}
  podLabels: {}
  replicaCount: 1
  resources: {}
  strategy:
    rollingUpdate:
      maxSurge: 50%
      maxUnavailable: 0
    type: RollingUpdate
  tolerations: []

HOOKS:
MANIFEST:
---
# Source: karmada-operator/templates/serviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: karmada-operator-operator
  namespace: karmada-system
  labels:
    app.kubernetes.io/name: karmada-operator
    helm.sh/chart: karmada-operator-0.0.0
    app.kubernetes.io/instance: karmada-operator
    app.kubernetes.io/managed-by: Helm
---
# Source: karmada-operator/templates/clusterrole.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: karmada-operator
  labels:
    app.kubernetes.io/name: karmada-operator
    helm.sh/chart: karmada-operator-0.0.0
    app.kubernetes.io/instance: karmada-operator
    app.kubernetes.io/managed-by: Helm
rules:
  - apiGroups: ['*']
    resources: ['*']
    verbs: ["*"]
  - nonResourceURLs: ['*']
    verbs: ["get"]
---
# Source: karmada-operator/templates/clusterrolebinding.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: karmada-operator
  labels:
    app.kubernetes.io/name: karmada-operator
    helm.sh/chart: karmada-operator-0.0.0
    app.kubernetes.io/instance: karmada-operator
    app.kubernetes.io/managed-by: Helm
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: karmada-operator
subjects:
  - kind: ServiceAccount
    name: karmada-operator-operator
    namespace: karmada-system
---
# Source: karmada-operator/templates/karmada-operator-deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: karmada-operator-operator
  namespace: karmada-system
  labels:
    app.kubernetes.io/name: karmada-operator
    helm.sh/chart: karmada-operator-0.0.0
    app.kubernetes.io/instance: karmada-operator
    app.kubernetes.io/managed-by: Helm
    app: karmada-operator-operator
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: karmada-operator
      app.kubernetes.io/instance: karmada-operator
      app: karmada-operator-operator
  strategy:
    rollingUpdate:
      maxSurge: 50%
      maxUnavailable: 0
    type: RollingUpdate
  template:
    metadata:
      labels:
        app.kubernetes.io/name: karmada-operator
        helm.sh/chart: karmada-operator-0.0.0
        app.kubernetes.io/instance: karmada-operator
        app.kubernetes.io/managed-by: Helm
        app: karmada-operator-operator
    spec:      
      containers:
      - name: karmada-operator-operator
        image: docker.io/karmada/karmada-operator:latest
        imagePullPolicy: IfNotPresent
        command:
        - /usr/local/bin/clustersynchro-manager
        - --leader-elect-resource-namespace=karmada-system
        - --v=2
      serviceAccountName: karmada-operator-operator

NOTES:
Thank you for installing karmada-operator.

Your release is named karmada-operator.

To learn more about the release, try:

  $ helm status karmada-operator -n karmada-system
  $ helm get all karmada-operator -n karmada-system
```  

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: charts/karmada-operator/values.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  ## @param image.registry operator image registry
+  ## @param image.repository operator image repository
+  ## @param image.tag operator image tag (immutable tags are recommended)
+  ## @param image.pullPolicy operator image pull policy
+  ## @param image.pullSecrets Specify docker-registry secret names as an array
+  ##
+  image:
+    registry: docker.io
+    repository: karmada/karmada-operator
+    tag: latest
```
Comment (liangyuanpeng) [inline charts/karmada-operator/values.yaml:33]: I think use latest tag is not a good idea.

**Review by liangyuanpeng — COMMENTED** [review]
Just add some comment.

1. Could you add CI for it ?
2.  I think `charts/karmada-operator/charts/common-1.17.1.tgz`  is not need to commit?

File: charts/karmada-operator/values.yaml (hunk 1 above)
Comment (calvin0327) [reply charts/karmada-operator/values.yaml:33]: thanks for your review. 
Now, we only have the latest version. 

the pkg `charts/karmada-operator/charts/common-1.17.1.tgz` is not need to commit indeed.

**Comment by calvin0327** [comment]
@Poor12 PTAL

File: charts/karmada-operator/values.yaml (hunk 1 above)
Comment (liangyuanpeng) [reply charts/karmada-operator/values.yaml:33]: > CI what you say is chart lint?

Yeah, samething of karmada helm chart. ref:
- https://github.com/karmada-io/karmada/blob/master/.github/workflows/lint-chart.yaml

Comment (Poor12) [reply charts/karmada-operator/values.yaml:33]: I think we can hold a while for it. We can improve it separately in the future.

**Comment by Poor12** [comment]
I think we need a readme for it. 
BTW, I use the command in the PR description and the deploy name is `karmada-operator-operator`.

**Comment by calvin0327** [comment]
> I think we need a readme for it. BTW, I use the command in the PR description and the deploy name is `karmada-operator-operator`.

@Poor12  ok , I added the readme doc in the PR. then I delete the suffix of operator deployment.

File: charts/karmada-operator/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# karmada-operator
```
Comment (Poor12) [inline charts/karmada-operator/README.md:1]: I'd like to invite @windsonsea to review the doc.

File: charts/karmada-operator/templates/karmada-operator-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        {{- include "common.tplvalues.render" (dict "value" .Values.operator.podLabels "context" $) | nindent 8 }}
+        {{- end }}
+    spec:
+      {{- include "karmada.operator.imagePullSecrets" . | indent 6 }}
+      containers:
+      - name: {{ include "karmada.operator.fullname" . }}
+        image: {{ template "karmada.operator.image" . }}
+        imagePullPolicy: {{ .Values.operator.image.pullPolicy }}
+        command:
+        - /bin/karmada-operator
```
Comment (Poor12) [inline charts/karmada-operator/templates/karmada-operator-deployment.yaml:40]: I find that we provided the option of kubeconfig before, do we need it in the chart?

File: charts/karmada-operator/README.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+The Karmada operator is a method for installing, upgrading, and deleting Karmada instances. It builds upon the basic Karmada resource and controller concepts, provides convenience to centrally manage entire lifecyle of Karmada instances in a global cluster. With the operator, you can extend Karmada with custom resources (CRs) to manage your instances not only in local
+clusters but also in remote clusters.
+
+## TL;DR
+
+Switch to the `root` directory of the repo.
+
+```console
+helm install karmada-operator -n karmada-system --create-namespace --dependency-update ./charts/karmada-operator
```
Comment (windsonsea) [inline charts/karmada-operator/README.md:13]: This command on line 13 has 90% similarity with that on line 31. Could we add more description?

File: charts/karmada-operator/README.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+helm install karmada-operator -n karmada-system  --create-namespace ./charts/karmada-operator --debug
+```
+
+```console
+kubectl get po -n karmada-system
+NAME                               READY   STATUS    RESTARTS   AGE
+karmada-operator-df95586f6-d5bll   1/1     Running   0          6s
+```
+
+## Uninstalling the Chart
```
Comment (windsonsea) [inline charts/karmada-operator/README.md:40]: Seems some content is missing between install and uninstall. You know, the purpose of installation is not to uninstall 🥲 .

What's the next after installing charts? only uninstall?

File: charts/karmada-operator/README.md (hunk 4 above)
Comment (calvin0327) [reply charts/karmada-operator/README.md:13]: 👌

File: charts/karmada-operator/README.md (hunk 5 above)
Comment (calvin0327) [reply charts/karmada-operator/README.md:40]: upgrade? I add upgrading content with installing together

File: charts/karmada-operator/README.md (hunk 2 above)
Comment (windsonsea) [reply charts/karmada-operator/README.md:1]: This README looks good to me now.

**Comment by calvin0327** [comment]
@RainbowMango /cc

**Review by RainbowMango — APPROVED** [review]
/approve
For the hack/ and .gitignore.

**Pull request merged**